geomesa-gt/geomesa-gt-partitioning/src/main/scala/org/locationtech/geomesa/gt/partition/postgis/dialect/tables/PrimaryKeyTable.scala

```diff
@@ -0,0 +1,49 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.gt.partition.postgis.dialect
+package tables
+
+import org.geotools.jdbc.MetadataTablePrimaryKeyFinder
+
+import java.util.Locale
+
+/**
+ * Primary key table used by the JDBC data store to specify primary key columns
+ */
+object PrimaryKeyTable extends Sql {
+
+  override def create(info: TypeInfo)(implicit ex: ExecutionContext): Unit = {
+    // we need to define the primary key separately since the main view can't have any primary key columns
+    val table = s""""${info.schema}"."${MetadataTablePrimaryKeyFinder.DEFAULT_TABLE.toLowerCase(Locale.US)}""""
+    val create =
+      s"""CREATE TABLE IF NOT EXISTS $table (
+         |  table_schema character varying,
+         |  table_name character varying,
+         |  pk_column_idx integer,
+         |  pk_column character varying,
+         |  pk_policy character varying,
+         |  pk_sequence character varying
+         |);""".stripMargin
+    val cleanup = s"""DELETE FROM $table WHERE table_schema = ? AND table_name = ?;"""
+    val entry = s"""INSERT INTO $table(table_schema, table_name, pk_column_idx, pk_column) VALUES (?, ?, ?, ?);"""
+    ex.execute(create)
+    ex.executeUpdate(cleanup, Seq(info.schema, info.tables.view.name.raw))
+    ex.executeUpdate(entry, Seq(info.schema, info.tables.view.name.raw, 0, "fid"))
```

Comment:
Is there any kind of locking which needs to happen to make sure these steps/actions happen "together" / atomically?

Reply:
they're part of a transaction so that should happen automatically. but this table is just modified up front during createSchema, so shouldn't really have any conflicts.

Reply:
Is there any locking on createSchema to prevent multiple clients from calling create at the same time?  Same with simultaneous calls to createSchema and deleteSchema?

Reply:
i don't think there's any explicit locking in java. due to standard sql transactions, i think things would generally be fine. whichever call gets in first will succeed, and the other would end up blocked and then fail and rollback. it would be the same as calling `create table foo` from 2 different sessions, which sql databases should handle fine.